File: top-interview-150/python/189_rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nums_size = len(nums)
        k = k % nums_size
        if k !=0 or k != nums_size:
            tmp = nums[:(nums_size-k)]
            nums[:k-1] = nums[(nums_size-k):]
            nums[k:] = tmp

File: top-interview-150/c++/189_rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int nums_size = nums.size();
        if (k > nums_size) k = k % nums_size;
        if (k != 0 | k != nums_size) {
            vector<int> tail{nums.begin() + nums_size-k, nums.end()};
            nums.insert(nums.begin(), tail.begin(), tail.end());
            nums.erase(nums.end()-k, nums.end());
        }
    }
};